### PR TITLE
Funtions

### PR DESCRIPTION
**Pull request by SonalPatel**
Firstly I created a function named "add"  to add two numbers together, Then declared the variable called "result" and passed in the parameters as numbers. Next I used the console to check my function was correct. Then called the variable "result" which added the two numbers together and output the result in an alert box. 

Then I converted this function into a fat arrow function.

### CODE REVIEW

**Comment by SonalPatel** [comment]
Function created to subtract two numbers using fat arrow function

**Comment by SonalPatel** [comment]
created a function to subtract two numbers

**Review by httpete-ire — COMMENTED** [review]
Looks good so far, have you done the other functions

File: functions/greater-than-ten/greater.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  } else {
+    return "sorry low number";
+  }
+}
+highNumber = 12;
+console.log(highNumber);
+alert(highNumber);
+*/
+
+checkValue = x => {
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: missing variable assignment ie `const checkValue = ...`

File: functions/greater-than-ten/greater.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  if (x > 10) {
+    return "15 is greater than 10";
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: function should return if the number is greater then 10, ie a boolean (true | flase)

File: functions/longest-word/long.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+//given two strings return the one with the longest word
+const nameSonal = "sonal";
+const NamePete = "pete";
+
+document.write("nameSonal.length is: " + nameSonal.length);
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: This isnt a reusable function, define a function that returns the longest word

File: functions/string-is-the-same/string.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//check if one string equals another.
+
+const s = "123";
```
Comment (httpete-ire) [inline functions/string-is-the-same/string.js]: create a reusable function

File: functions/longest-word/long.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+//given two strings return the one with the longest word
+const nameSonal = "sonal";
+const namePete = "pete";
+
+function findlongestword() {
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: you should be taking in the names as parameters

File: functions/longest-word/long.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+//given two strings return the one with the longest word
+const nameSonal = "sonal";
+const namePete = "pete";
+
+function findlongestword() {
+  if (nameSonal.length > namePete.length) {
+    alert(nameSonal);
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: should be returning the name

File: functions/longest-word/long.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+//given two strings return the one with the longest word
+const nameSonal = "sonal";
+const namePete = "pete";
+
+function findlongestword() {
+  if (nameSonal.length > namePete.length) {
+    alert(nameSonal);
+  } else {
+    alert(namePete);
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: same as above

File: functions/greater-than-ten/greater.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+checkValue = x => {
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: need to assign this to a variable

File: functions/greater-than-ten/greater.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+checkValue = x => {
+  if (x > 10) {
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: the function should return if the string is greater then ten, you are not returning a string it should be a boolean

File: functions/greater-than-ten/greater.js (hunk 8 above)
Comment (httpete-ire) [reply functions/greater-than-ten/greater.js]: I would use a more descriptive name then x

File: functions/greater-than-ten/greater.js (hunk 1 above)
Comment (SonalPatel) [reply functions/greater-than-ten/greater.js]: amended the code

File: functions/greater-than-ten/greater.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+const checkValue = x => {
+  if (x > 10) {
+    return "true";
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: return a string instead of a boolean

Comment (httpete-ire) [reply functions/greater-than-ten/greater.js]: return true | false

File: functions/longest-word/long.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const nameOne = "Sonal is learning js";
+const nameTwo = "Pete is already awesome at js";
+
+function findlongestword(sonal, pete) {
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: I would name the parameters `name1, name2`

File: functions/longest-word/long.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const nameOne = "Sonal is learning js";
+const nameTwo = "Pete is already awesome at js";
+
+function findlongestword(sonal, pete) {
+  if (nameOne.length > nameTwo.length) {
+    alert(nameOne);
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: need to return instead of alerting the name

File: functions/longest-word/long.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const nameOne = "Sonal is learning js";
+const nameTwo = "Pete is already awesome at js";
+
+function findlongestword(sonal, pete) {
+  if (nameOne.length > nameTwo.length) {
+    alert(nameOne);
+  } else {
+    alert(nameTwo);
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: same as above

File: functions/string-is-the-same/string.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+  alert(true);
+} else {
+  alert(false);
+}
+*/
+
+//now re-create this as a reusable function
+const s = "123";
+const p = "321";
+function reUsable(s, p) {
```
Comment (httpete-ire) [inline functions/string-is-the-same/string.js]: rename to `isTheSameWord`

File: functions/string-is-the-same/string.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  if (s === p) {
+    return true;
+  } else {
+    return false;
+  }
+}
+alert(reUsable(p));
```
Comment (httpete-ire) [inline functions/string-is-the-same/string.js]: only passing in one value

File: functions/string-is-the-same/string.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+
+if (s === p) {
+  alert(true);
+} else {
+  alert(false);
+}
+*/
+
+//now re-create this as a reusable function
+const s = "123";
```
Comment (httpete-ire) [inline functions/string-is-the-same/string.js]: remove s and p, I think they are confusing you

File: functions/longest-word/long.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const name1 = "Sonal is learning js";
+const name2 = "Pete is already awesome at js";
+
+function findlongestword(x, y) {
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: rename `x` and `y` too `name1` and `name2`

File: functions/longest-word/long.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const name1 = "Sonal is learning js";
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: remove these as they are confusing

File: functions/longest-word/long.js [hunk 19]
```diff
@@ -0,0 +1,12 @@
+//given two strings return the one with the longest word
+const name1 = "Sonal is learning js";
+const name2 = "Pete is already awesome at js";
+
+function findlongestword(x, y) {
+  if (name2.length > name2.length) {
```
Comment (httpete-ire) [inline functions/longest-word/long.js]: checking length `name2` is greater  then `name2`

Comment (httpete-ire) [reply functions/longest-word/long.js]: `name2.length > name1.length`

File: functions/greater-than-ten/greater.js [hunk 20]
```diff
@@ -0,0 +1,11 @@
+const checkValue = x => {
+  if (x > 10) {
+    return "15 is greater than 10";
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: return true

File: functions/greater-than-ten/greater.js [hunk 21]
```diff
@@ -0,0 +1,11 @@
+const checkValue = x => {
+  if (x > 10) {
+    return "15 is greater than 10";
+    //executes if true
+  } else {
+    return "ooops";
```
Comment (httpete-ire) [inline functions/greater-than-ten/greater.js]: return false

**Review by httpete-ire — APPROVED** [review]
LGTM

**Pull request merged**